runtime/sema/type.go

```diff
@@ -7987,10 +7987,6 @@ type IntersectionType struct {
 var _ Type = &IntersectionType{}
 
 func NewIntersectionType(memoryGauge common.MemoryGauge, types []*InterfaceType) *IntersectionType {
-	if len(types) == 0 {
-		panic(errors.NewUnreachableError())
-	}
```

Comment:
@onflow/cadence was there a particular reason why we were forbidding empty intersection types?

They are not useful, but I cannot think of a reason why they should not be allowed.

They might be a result of migrating a restricted type without any restrictions.

Reply:
I guess we cannot properly determine resource-kindedness for an empty intersection?